ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/AnimationFrameCache.cs

```diff
@@ -35,6 +41,10 @@ public void Clear()
         /// <param name="time">The time to get the frame for.</param>
         public Frame GetInterpolatedFrame(Animation anim, float time)
         {
+            if (lastAnimation == anim && Math.Abs(lastTime - time) < 0.0001)
```

Comment:
What is this for?

Reply:
ModelSceneNode.Update calls AnimationController.GetAnimationMatrices and AnimationController.GetDatas, which both call this function. I thought it would make sense that it returns a cached frame if it's called for the same frame, given that the class already stores the interpolated frame without doing much with it, and it being called AnimationFrameCache.

Reply:
I don't like this kind of code, it means we don't have full grasp of the flow here, ideally getting a frame during update would be called only once.

Reply:
I've pushed a different implementation just now